### PR TITLE
Itinerary

### PR DESCRIPTION
**Pull request by jlanson**
Refactor openweather and places api
itinerary items data structure includes start/endDate
Can Add items with calendar + time picker
Serach functionality 
Can edit items 
UI updates

### CODE REVIEW

File: app/src/main/java/com/example/packpals/repositories/ItineraryRepository.kt [hunk 1]
```diff
@@ -1,72 +1,65 @@
...
+    ) {
+
 
     suspend fun fetchItems(tripId: String): List<Itinerary_Item>? {
         val tripFilter = Filter.equalTo("tripId", tripId)
         var itineraryItems = emptyList<Itinerary_Item>()
+        val newItineraryItems = mutableListOf<Itinerary_Item>()
+        val newItineraryItemsWithId = mutableListOf<Itinerary_Item>()
+
+        itineraryItems = itineraryCollectionRef.where(tripFilter).get().await().toObjects(Itinerary_Item::class.java)
```
Comment (alex-fang0) [inline app/src/main/java/com/example/packpals/repositories/ItineraryRepository.kt]: We do we need 2 seperate calls for getting itinerary items? This one seems unused

File: app/src/main/java/com/example/packpals/repositories/OpenWeatherRepository.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import java.net.URL
+import javax.inject.Inject
+
+class OpenWeatherRepository @Inject constructor(){
+
+    //Todo: Dont hardcode
```
Comment (stephchi0) [inline app/src/main/java/com/example/packpals/repositories/OpenWeatherRepository.kt:17]: uh..

File: app/src/main/java/com/example/packpals/views/itinerary/AddItineraryFragment.kt [hunk 3]
```diff
@@ -38,19 +41,26 @@ class AddItineraryItemFragment : Fragment(){
...
-//                itineraryView.findViewById<TextView>(R.id.tvdate).text = date
-//                itineraryView.findViewById<TextView>(R.id.tvforecast).text = forecast
-                itineraryView.findViewById<ImageView>(R.id.image).setImageResource(R.mipmap.fenugs)
-
-                itineraryView.setOnClickListener {
-                    // fill in later
-                }
-
-                linearLayout.addView(itineraryView)
+//            for (item in itineraryItems) {
```
Comment (stephchi0) [inline app/src/main/java/com/example/packpals/views/itinerary/AddItineraryFragment.kt:47]: so is this code just not going to be used?

**Review by stephchi0 — COMMENTED** [review]
clean up the commented stuff ig

**Review by alex-fang0 — APPROVED** [review]
Other than that comment lgtm

**Pull request merged**